dmvutil.py

```diff
@@ -0,0 +1,253 @@
+        if len(l) > 0:
+            idx = l.index(max(l))
+            variant = self.variants[idx]
+        return variant
+
+    def getSelectedText(self):
+        if self.selected != None:
+            return self.selected
+        return "N/A"
+
+    def getActiveText(self):
+        if self.active != None:
+            return self.active
+        return "N/A"
+
+def readPciDriver(pci_id):
+    driver_path = f"/sys/bus/pci/devices/{pci_id}/driver"
+    if os.path.exists(driver_path):
+        driver_name = os.path.basename(os.path.realpath(driver_path))
+        return driver_name
+    # some devices like ACPI, PCI bridge
+    return None
+
+def filterDeviceRcv(pci_device):
+    filtered_text = re.sub(r'\(rev\s+\w+\)', '', pci_device).strip()
+    return filtered_text
+
+def getDeviceDriverMap(pci_dev_list):
+    device_driver_map = {}
+    for pci_dev in pci_dev_list:
+        driver = pci_dev.driver
+        if driver in device_driver_map:
+            l = device_driver_map[driver]
+            if pci_dev.getHumanDevLabel() not in l:
+                l.append(pci_dev.getHumanDevLabel())
+            device_driver_map[driver] = l
+        else:
+            device_driver_map[driver] = [pci_dev.getHumanDevLabel()]
+    return device_driver_map
+
+def parsePCIData(pcilist):
+    pci_dev_list = []
+    for device in pcilist.split("\n"):
+        pattern = re.compile(
+            r'^(?P<slot>\S+)\s+(?P<class>.+?):\s+(?P<vendor>.+?)\s+(?P<device>.+?)(?: \((?P<rev>\w+)\))?$')
+        for match in pattern.finditer(device):
+            pci_info = match.groupdict()
+            pci_slot = pci_info["slot"]
+            devclass = pci_info["class"]
+            vendor_id = pci_info["vendor"]
+            device_id = filterDeviceRcv(pci_info["device"])
+            rev = pci_info["rev"]
+
+            pci_id = "0000:" + pci_slot
+            driver = readPciDriver(pci_id)
+            if driver:
+                pci_dev = PciDevice(pci_id, devclass, vendor_id, device_id, driver)
+                pci_dev_list.append(pci_dev)
+    return getDeviceDriverMap(pci_dev_list)
+
+def parseDMVJsonData(dmvlist, device_driver_map):
+    def sortKey(e):
+        return e.type
+
+    drivers = []
+    for name, info in dmvlist.items():
+        drivers.append(Drivers(name, info))
+    drivers.sort(key=sortKey)
+
+    for drvname, devlist in device_driver_map.items():
+        for driver in drivers:
+            if driver.drvname == drvname:
+                driver.setDeviceList(devlist)
+    return drivers
+
+def cloneDriverWithoutVariants(olddrv, newdrv):
+    newdrv.type = olddrv.type
+    newdrv.friendly_name = olddrv.friendly_name
+    newdrv.description = olddrv.description
+    newdrv.info = olddrv.info
+    newdrv.selected = olddrv.selected
+    newdrv.active = olddrv.active
+
+def getHardwarePresentDrivers(drivers):
+    hw_present_drivers = []
+    for d in drivers:
+        driver = Drivers(d.drvname)
```

Comment:
Why not change the `Drivers()` constructor so that it can take a list of variants and pass the filtered list below?

That way you don't have to construct a Drivers object with variants, copy it and throw the variants away, then set a new list of variants.

(Would there be any variants to throw away anyway since you haven't passed in driver info?)